src/core/agent/services/ipexCommunicationService.ts

```diff
@@ -658,6 +652,217 @@ class IpexCommunicationService extends AgentService {
       await this.notificationStorage.update(notificationRecord);
     }
   }
+
+  async multisigOfferAcdcFromApply(
+    multisigId: string,
+    notificationSaid: string,
+    acdcDetail: any,
+    discloseePrefix: string,
+    offerExnToJoin?: any
+  ) {
+    let exn: Serder;
+    let sigsMes: string[];
+    let dtime: string;
+
+    const { ourIdentifier, multisigMembers } =
+      await this.multisigService.getMultisigParticipants(multisigId);
+
+    const gHab = await this.props.signifyClient.identifiers().get(multisigId);
+    const mHab = await this.props.signifyClient
+      .identifiers()
+      .get(ourIdentifier.id);
+
+    const recp = multisigMembers
+      .filter((signing: any) => signing.aid !== ourIdentifier.id)
+      .map((member: any) => member.aid);
+
+    const [_, acdc] = Saider.saidify(acdcDetail);
+
+    if (offerExnToJoin) {
+      const [, ked] = Saider.saidify(offerExnToJoin);
+      const offer = new Serder(ked);
+      const keeper = await this.props.signifyClient.manager!.get(gHab);
+      const sigs = await keeper.sign(b(new Serder(offerExnToJoin).raw));
+
+      const mstateNew = gHab["state"];
+      const seal = [
+        "SealEvent",
+        {
+          i: gHab["prefix"],
+          s: mstateNew["ee"]["s"],
+          d: mstateNew["ee"]["d"],
+        },
+      ];
+      const signer = sigs.map((sig: any) => new Siger({ qb64: sig }));
+      const ims = d(messagize(offer, signer, seal));
+      const atc = ims.substring(offer.size);
+
+      const gembeds = {
+        exn: [offer, atc],
+      };
+
+      [exn, sigsMes, dtime] = await this.props.signifyClient
+        .exchanges()
+        .createExchangeMessage(
+          mHab,
+          MultiSigRoute.EXN,
+          { gid: gHab["prefix"] },
+          gembeds,
+          discloseePrefix
+        );
+    } else {
+      const time = new Date().toISOString().replace("Z", "000+00:00");
+      const exchangeMessage = await this.props.signifyClient
+        .exchanges()
+        .get(notificationSaid);
+
+      const applySaid = exchangeMessage.exn.d;
```

Comment:
We already have the notification in `offerAcdcFromApply` - can we just pass the notification through instead of `notificationSaid`? Since `applySaid` is in the notification object in the `a.r` field.

(`applyNoteRecord`)

Reply:
In general, the less calls to KERIA we can do the better since it's over the internet - and every request must be signed (and soon encrypted) which takes time

Reply:
I removed the get `notification` here. But I still used `notificationSaid`. I noticed that the joinMultisigOffer() function is also using it and passing notification as a parameter might not be appropriate in that case so I decided to keep `notificationSaid` instead of `notification`. However, joinMultisigOffer() doesn't need to pass `notificationSaid` or `notification` since the information is mainly retrieved through `multisigExn`. So I'll leave this field optional.

Reply:
yep sorry my mistake